src/commondao/common.dao.ts

```diff
@@ -715,7 +715,8 @@ export class CommonDao<
     const dbms = await this.bmsToDBM(bms, opt)
     if (opt.ensureUniqueId) throw new AppError('ensureUniqueId is not supported in saveBatch')
     if (this.cfg.immutable)
-      throw new AppError('immutable DB entries are not supported in saveBatch')
+      await pMap(dbms, async dbm => await this.ensureImmutableDoesntExist(table, dbm))
```

Comment:
Just flagging that it's slower to do multiple `getById` calls (as it'll happen now) versus doing one `getByIds` call with the batch of ids.

I recommend to do that.

Reply:
Good point, fixed